platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/internal/services/ToolchainsJvmServices.java

```diff
@@ -73,8 +72,8 @@ public class ToolchainsJvmServices extends AbstractGradleModuleServices {
     protected static class BuildServices implements ServiceRegistrationProvider {
 
         @Provides
-        protected BuildPlatform createBuildPlatform(ObjectFactory objectFactory, SystemInfo systemInfo, OperatingSystem operatingSystem) {
-            return objectFactory.newInstance(DefaultBuildPlatform.class, systemInfo, operatingSystem);
+        protected CurrentBuildPlatform createBuildPlatform(ObjectFactory objectFactory, SystemInfo systemInfo, OperatingSystem operatingSystem) {
```

Comment:
❓  Why does the service still need to be created via an `ObjectFactory`?

Maybe previously it had some abstract getters that were injected? Currently it seems we can plainly call the constructor, removing the need to have the `ObjectFactory` here.

Reply:
The `SystemInfo` and `OperatingSystem` seem to be available in the `Global` scope (from `NativeServices`)

It means the `CurrentBuildPlatform` can become a global service as well.

Reply:
Indeed, that makes sense so moved to `Global` scope.